market_maker_keeper/band.py

```diff
@@ -415,6 +414,17 @@ def _new_buy_orders(self, our_buy_orders: list, our_buy_balance: Wad, target_pri
 
         return new_orders, missing_amount
 
+    def _calculate_price(self, band: Band, target_price: Wad) -> Wad:
+        price = band.avg_price(target_price)
+
+        return price
+
+    def _calculate_pay_and_buy_amounts(self, band: Band, price: Wad, our_balance: Wad, total_amount: Wad, limit_amount: Wad) -> Tuple[Wad, Wad, Wad]:
+        pay_amount = Wad.min(band.avg_amount - total_amount, our_balance, limit_amount)
```

Comment:
Buy amount calculations don't match existing logic for but and sell (one is *, one is /). Also, there's an unnecessary extra `Wad` in the return Tuple. It would probably be better to just add additional logic to BinanceBands as opposed to modifying Bands.py which require retesting our existing keepers. Any errors here could be catastrophic.

Reply:
+1 to this ^^ lets create a new class vs modifying.

Reply:
Sure, the easiest way to achieve this in BinanceBands is to override `_new_buy_orders` and `_new_sell_orders` with copied functionality. 

Another way, that modifies `band.py` is to add `_calculate_buy_amount_for_sell_orders` and `_calculate_buy_amount_for_buy_orders` which will be one line methods.